haas/api.py

```diff
@@ -263,6 +263,19 @@ def node_power_off(node):
     node.obm.power_off()
 
 
+@rest_call('PUT', '/node/<node>/boot_device', Schema({
+    'node': basestring, 'bootdev': basestring,
+}))
+def node_set_bootdev(node, bootdev):
+    auth_backend = get_auth_backend()
+    node = _must_find(model.Node, node)
+    if node.project is None:
+        auth_backend.require_admin()
+    else:
+        auth_backend.require_project_access(node.project)
+    node.obm.set_bootdev(bootdev)
```

Comment:
One idea that Naved had is to validate the valid bootdev here and not in the driver by adding a callout like `validatte_bootdev` or something.

I really like that, as it keeps invalid values out of the lower depths.

Another option I thought of would be to allow extensions to modify bootdev, but that doesn't work because you could have multiple drivers that support different bootdevs, and a valid bootdev on one (like dell_idrac) wouldn't necessarily work on another (like ipmi).